deps/src/singular.cpp

```diff
@@ -53,6 +75,46 @@ JLCXX_MODULE define_julia_module(jlcxx::Module & Singular)
     singular_define_ideals(Singular);
     singular_define_matrices(Singular);
 
+    Singular.method("call_interpreter", [](std::string input) {
+        // set callbacks
+        auto default_print = PrintS_callback;
+        auto default_error = WerrorS_callback;
+        auto default_warning = WarnS_callback;
+
+        PrintS_callback = PrintS_for_julia;
+        WerrorS_callback = WerrorS_for_julia;
+        WarnS_callback = WarningS_for_julia;
+
+        // cleanup return strings
+        singular_return.erase();
+        singular_error.erase();
+        singular_warning.erase();
+
+        // call interpreter
+        std::string  input_str = input + "\nreturn();";
+        const char * input_string = input_str.c_str();
+        char *       ost = (char *)omalloc0(strlen(input_string) + 1);
+        strcpy(ost, input_string);
+        bool err = iiAllStart(NULL, ost, BT_proc, 0);
+        inerror = 0;
+        errorreported = 0;
+        omFree(ost);
```

Comment:
Hmm, if you are responsible for freeing `ost` -- then does it even matter how it was allocated? Can't you just pass `input_str.c_str()` to `iiAllStart`?

Reply:
I am not completely sure about this. I could try, though. But maybe Singular likes it to be allocated via `omalloc`

Reply:
I had a look at the code for `iiAllStart`. It does exactly one thing with that argument: it passes it to `omStrDup` :-). So, yeah, I am pretty sure that you can just pass `input_str.c_str()` to `iiAllStart` :-)